src/figuresClasses.ts

```diff
@@ -1,11 +1,78 @@
-export interface Figure {}
+export type Shape = 'triangle' | 'circle' | 'rectangle';
+export type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+function checkOnError(errorMessage: string, ...args: number[]): void {
+  if (args.some((arg: number) => arg <= 0)) {
+    throw new Error(errorMessage);
+  }
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - The constructor parameters `color`, `a`, `b`, and `c` are being manually assigned to class properties, but this is unnecessary because TypeScript can do this automatically when using the `public` keyword in the constructor parameters. You can simplify the constructor by removing the property assignments.